Added single stream read/write, callback capability

wherrera10 opened the pull request:
This adds functionality and functions similar to the Python PyAudio interface to the PortAudio library.

ssfrr commented:
As I mentioned in #51, I added some more commits on top of your changes in that PR and pushed to the `wherrera_10_test` branch, most importantly getting the tests working again, including travis testing.

Would you mind rebasing this onto that branch? Currently I can't run the tests as-is on this PR (are they working for you?). Also, most of the commits here are just the name of the file that was created or updated. Can you squash down to fewer and more descriptive commits? It makes it a lot easier for me to see what's going on.

Thanks a lot! I'm looking forward to digging into these changes and getting this merged.

wherrera10 commented:
It seems that I cannot get the GitHub web site to do a rebase, so I deleted
the fork, reforked master, and changed portaudio.jl, README.md, and
build.jl plus added the example files. GitHub it seems does not like trying
to fork your forks, it wants to fork from master only.

I'll hold off on sending a pull request til you ask though. I have audio
working, which means that I can believe in what Julia does with streaming
data in general. So I will be able to get back to the actual
work-in-progress, which is with much lower frequency, multiband EDF data
file stream files, not audio really.

--Bill


On Thu, Dec 3, 2015 at 10:03 AM, Spencer Russell <notifications@github.com>
wrote:

> As I mentioned in #51 <https://github.com/ssfrr/AudioIO.jl/pull/51>, I
> added some more commits on top of your changes in that PR and pushed to the
> wherrera_10_test branch, most importantly getting the tests working
> again, including travis testing.
>
> Would you mind rebasing this onto that branch? Currently I can't run the
> tests as-is on this PR (are they working for you?). Also, most of the
> commits here are just the name of the file that was created or updated. Can
> you squash down to fewer and more descriptive commits? It makes it a lot
> easier for me to see what's going on.
>
> Thanks a lot! I'm looking forward to digging into these changes and
> getting this merged.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ssfrr/AudioIO.jl/pull/55#issuecomment-161767165>.
>

ssfrr commented:
Does your new PR work off of the `wherrera10_test` branch that gets the unit tests working? That's the main thing. Secondarily having a smaller number of commits with descriptive names keeps the repo history more tidy and easier to manage.

Also, as an aside - are you doing all your git interaction through the GitHub web interface? If so I'd recommend using a local git client (I hear GitHub Desktop is quite good), as there are lots of things you can't do from the website (rebasing, squashing commits, putting together multi-file commits).

wherrera10 commented:
I guess I cannot easily get my Github program to clone and fork a fork-- it
wants master.

I did compare my revised file to the wherrera10 branch and found two places
where the wherrera10 had an update in portaudio.jl and added those to the
branch here, so a pull should work. I guess you are working on a
substantial revision though?




On Wed, Dec 9, 2015 at 4:55 AM, Spencer Russell <notifications@github.com>
wrote:

> Does your new PR work off of the wherrera10_test branch that gets the
> unit tests working? That's the main thing. Secondarily having a smaller
> number of commits with descriptive names keeps the repo history more tidy
> and easier to manage.
>
> Also, as an aside - are you doing all your git interaction through the
> GitHub web interface? If so I'd recommend using a local git client (I hear
> GitHub Desktop is quite good), as there are lots of things you can't do
> from the website (rebasing, squashing commits, putting together multi-file
> commits).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ssfrr/AudioIO.jl/pull/55#issuecomment-163279070>.
>
